### PR TITLE
switch to message only predicate

### PR DESCRIPTION
**Pull request by adamralph**
Relates to https://github.com/adamralph/bullseye/issues/172

### CODE REVIEW

**Comment by adamralph** [comment]
Kudos to @thomaslevesque for this suggestion!

**Comment by adamralph** [comment]
Released in [beta 5](https://www.nuget.org/packages/Bullseye/2.3.0-beta.5).

File: Bullseye/Internal/TargetCollectionExtensions.cs [hunk 1]
```diff
@@ -30,13 +30,14 @@ private static async Task RunAndExitAsync(this TargetCollection targets, List<st
...
                 await RunAsync(targets, names, options, log, args).ConfigureAwait(false);
             }
-            catch (Exception ex) when (exceptionMessageOnly.Concat(new[] { typeof(BullseyeException) }).Any(type => type.IsAssignableFrom(ex.GetType())))
-            {
-                await log.Error(ex.Message).ConfigureAwait(false);
-                Environment.Exit(2);
-            }
             catch (Exception ex)
             {
+                if (typeof(BullseyeException).IsAssignableFrom(ex.GetType()) || messageOnly(ex))
```
Comment (thomaslevesque) [inline Bullseye/Internal/TargetCollectionExtensions.cs]: Sorry I'm late to the party, but why not just `ex is BullseyeException`?

Comment (adamralph) [reply Bullseye/Internal/TargetCollectionExtensions.cs]: Yeah, I guess that would have worked, but this all went away in https://github.com/adamralph/bullseye/pull/189

**Pull request merged**